src/fri/naive.md

```diff
@@ -0,0 +1,157 @@
+## How does FRI work?
+FRI protocol relies on a fact that if a source polynomial $p_0$ can be recursively folded into a constant polynomial, then $p_0$ has an expected upper bound degree. By folding, we mean each round halves the degree of $p_0$.
+
+Simply put, it is a protocol for verifier to quickly check if a polynomial has a upper degree bound claimed by prover.
+
+This post will explain the intuition of the math behind how polynomial folding works through some simple calculation examples end to end. These examples will demonstrate a naive version of FRI protocol without the the uses of commitment, which will explained in another post. 
+
+## What is folding?
+
+The core of the protocol relies on a technique to recursively fold the original polynomial and derive a constant value at the end. Here is an example:
+
+$$
+\begin{align*}
+p_0\text{ : } & 1 + 2x + 3x^2 & \\
+p_1\text{ : } & 3 + 3y & \\
+p_2\text{ : } & 6 &
+\end{align*}
+$$
+
+
+$p_0$ is the source polynomial. After a round of folding, it results in $p_1$. Then, the final round results in a constant polynomial $p_2$.
+
+To see how this folding algorithm works, please refer to the [code](https://github.com/katat/fri/blob/ab5aad54b8fd1e37b881ed7558d6ad22b6911442/src/poly.rs#L77).
+
+These folded polynomials have the following relationship with their source polynomials:
+
+$$p_i\left(x^2\right)=\frac{p_{i-1}\left(x\right)+p_{i-1}\left(-x\right)}{2}+\beta_i \frac{p_{i-1}\left(x\right)-p_{i-1}\left(-x\right)}{2 x}$$
+
+
+*For simplicity, the random number $\beta$ will be assumed as 1 in this post. We would touch the implication behind this random number in another post that explains the commitment.*
+
+$p_0$ is the original polynomial, and $p_2$ is a constant polynomial that can't be folded anymore. 
+
+These folded polynomials can be traced back to their source polynomials. Thus, the evaluations in a folded polynomial can be accumulated from the corresponding source polynomial. 
+
+These relationships can serve as a proof for verifier to the consistency among the evaluations. The consistency check implies if there is a $p_0$ exists that the prover claimed.
+
+With the same folding setup, the final polynomial will be always the same value. Verifier can check against the same final value against the reduced values from different evaluation points.
```

Comment:
what is a folding setup? you mean with the same polynomial and if the verifier sends the same randomness? I'm not sure I understand the paragraph here